app/javascript/controllers/concepts_json.js

```diff
@@ -0,0 +1,41 @@
+import {Controller} from "@hotwired/stimulus"
+
+export default class extends Controller {
+    static values = {
+        ontologyId: String,
+        conceptId: String,
+        apiKey: String
+      }
+
+    update(){
+        const baseClassUrl = `${this.ontologyIdValue}/classes/${this.conceptIdValue}`;
+        const apikey = `apikey=${this.apiKeyValue}`;
+        const tabsList = document.querySelectorAll('#concept_tabs_container .nav-item');
+        const links = {
+            details_tab: `${baseClassUrl}?${apikey}&display=all`,
+            instances_tab: `${baseClassUrl}/instances?${apikey}`,
+            notes_tab: `${baseClassUrl}/notes?${apikey}`,
+            mappings_tab: `${baseClassUrl}/mappings?${apikey}`,
+            visualization_tab: ''
+        };
+
+        let selectedItemId;
+        for (let i = 0; i < tabsList.length; i++) {
```

Comment:
change to 
```js
const selectedItemId = tabsList.find(tab => tab.classList.contains('active'))?.querySelector('a').id; 
```

Reply:
Not working, even chat gpt can't give a better version of this 😂

Reply:
It should work. Try to debug why not, this code needs to remain as this refactored version.